README
NAME

    Dist::Zilla::PluginBundle::Author::RTHOMPSON - RTHOMPSON's Dist::Zilla
    Configuration

VERSION

    version 0.142221

SYNOPSIS

    In dist.ini:

        [@Author::RTHOMPSON]

DESCRIPTION

    This plugin bundle, in its default configuration, is equivalent to:

        [AutoVersion]
        major = 0
        [GatherDir]
        [PruneCruft]
        [ManifestSkip]
        [MetaYAML]
        [MetaNoIndex]
        dir = corpus
        [License]
        [ExecDir]
        [ShareDir]
        [MakeMaker]
        [Manifest]
        [PkgVersion]
        [PodWeaver]
        [InstallGuide]
        [ReadmeAnyFromPod / ReadmeTextInBuild ]
        [ReadmeAnyFromPod / ReadmePodInRoot ]
        phase = release
        [Test::Perl::Critic]
        [PodCoverageTests]
        [PodSyntaxTests]
        [HasVersionTests]
        [Test::Portability]
        [Test::UnusedVars]
        [Test::Compile]
        skip = Test$
        [Test::Kwalitee]
        [ExtraTests]
        [ReportVersions]
        [MinimumPerl]
        [AutoPrereqs]
        [CheckChangesHasContent]
        [CheckPrereqsIndexed]
        [CheckVersionIncrement]
        [NextRelease]
        [TestRelease]
        [ConfirmRelease]
        [UploadToCPAN]
        [ArchiveRelease]
        directory = releases
        [Git::Check]
        allow_dirty = dist.ini
        allow_dirty = README.pod
        allow_dirty = Changes
        [Git::Commit]
        allow_dirty = dist.ini
        allow_dirty = README.pod
        allow_dirty = Changes
        [Git::Tag]
        [Git::Push]
        push_to = origin
        [Git::Remote::Check]
        remote_name = origin
        branch = master
        remote_branch = master
        [GithubMeta]

    There are several options that can change the default configuation,
    though.

OPTIONS

 -remove

    This option can be used to remove specific plugins from the bundle. It
    can be used multiple times.

    Obviously, the default is not to remove any plugins.

    Example:

        ; Remove these two plugins from the bundle
        -remove = Test::Perl::Critic
        -remove = GithubMeta

 version, version_major

    This option is used to specify the version of the module. The default
    is 'auto', which uses the AutoVersion plugin to choose a version
    number. You can also set the version number manually, or choose
    'disable' to prevent this bundle from supplying a version.

    Examples:

        ; Use AutoVersion (default)
        version = auto
        version_major = 0
        ; Use manual versioning
        version = 1.14.04
        ; Provide no version, so that another plugin can handle it.
        version = disable

 copy_file, move_file

    If you want to copy or move files out of the build dir and into the
    distribution dir, use these two options to specify those files. Both of
    these options can be specified multiple times.

    The most common reason to use this would be to put automatically
    generated files under version control. For example, Github likes to see
    a README file in your distribution, but if your README file is
    auto-generated during the build, you need to copy each newly-generated
    README file out of its build directory in order for Github to see it.

    If you want to include an auto-generated file in your distribution but
    you don't want to include it in the build, use move_file instead of
    copy_file.

    By default, both of these options are unset.

    Example:

        copy_file = README
        move_file = README.pod
        copy_file = README.txt

 synopsis_is_perl_code

    If this is set to true (the default), then the SynopsisTests plugin
    will be enabled. This plugin checks the perl syntax of the SYNOPSIS
    sections of your modules. Obviously, if your SYNOPSIS section is not
    perl code (case in point: this module), you should set this to false.

    Example:

        synopsis_is_perl_code = false

 release

    This option chooses the type of release to do. The default is 'real,'
    which means "really upload the release to CPAN" (i.e. load the
    UploadToCPAN plugin). You can set it to 'fake,' in which case the
    FakeRelease plugin will be loaded, which simulates the release process
    without actually doing anything. You can also set it to 'none' if you
    do not want this module to load any release plugin, in which case your
    dist.ini file should load a release plugin directly. Any other value
    for this option will be interpreted as a release plugin name to be
    loaded.

    Examples:

        ; Release to CPAN for real (default)
        release = real
        ; For testing, you can do fake releases
        release = fake
        ; Or you can choose no release plugin
        release = none
        ; Or you can specify a specific release plugin.
        release = OtherReleasePlugin

 archive, archive_directory

    If set to true, the archive option copies each released version of the
    module to an archive directory, using the ArchiveRelease plugin. This
    is the default. The name of the archive directory is specified using
    archive_directory, which is releases by default.

    Examples:

        ; archive each release to the "releases" directory
        archive = true
        archive_directory = releases
        ; Or don't archive
        archive = false

 vcs

    This option specifies which version control system is being used for
    the distribution. Integration for that version control system is
    enabled. The default is 'git', and currently the only other option is
    'none', which does not load any version control plugins.

 git_remote

    This option specifies the primary Git remote for the repository. The
    default is 'origin'. To disable all Git remote operations, set this to
    an empty string.

 git_branch, git_remote_branch

    This option specifies the branch that is to be checked against its
    remote. The default is 'master'. The second option, git_remote_branch,
    is only needed if the remote branch has a different name. It will
    default to being the same as git_branch.

 no_check_remote

    By default, the Git branch git_branch will be checked against the
    remote branch git_remote_branch at the remote specified by git_remote
    using the Git::Remote::Check plugin. If the remote branch is ahead of
    the local branch, the release process will be aborted. This option
    disables the check, allowing a release to happen even if the check
    would fail. This option has no effect if either git_remote or
    git_branch is set to an empty string.

 no_push

    By default, the Git repo will be pushed to the remote specified by
    git_remote after every release, to ensure that the remote repository
    contains the latest release. To disable pushing after a release, set
    this option. This option has no effect if git_remote is set to an empty
    string.

 allow_dirty

    This corresponds to the option of the same name in the Git::Check and
    Git::Commit plugins. Briefly, files listed in allow_dirty are allowed
    to have changes that are not yet committed to git, and during the
    release process, they will be checked in (committed).

    The default is dist.ini, Changes, and README.pod. If you override the
    default, you must include these files manually if you want them.

    This option only has an effect if vcs is 'git'.

 PLUGIN-SPECIFIC OPTIONS

    This bundle consumes the
    Dist::Zilla::Role::PluginBundle::Config::Slicer role, which means that
    you can specify any option to any plugin in the bundle directly by
    prefixing it with the plugin's name. This allows you to configure any
    options not covered by the above. For example, if you want to select
    "scripts" as the directory for the ExecDir plugin, you would specify
    the option as ExecDir.dir = "scripts".

BUGS AND LIMITATIONS

    This module should be more configurable. Suggestions welcome.

    Please report any bugs or feature requests to
    rct+perlbug@thompsonclan.org.

INSTALLATION

    See perlmodinstall for information and options on installing Perl
    modules.

AUTHOR

    Ryan C. Thompson <rct@thompsonclan.org>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2010 by Ryan C. Thompson.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.

DISCLAIMER OF WARRANTY

    BECAUSE THIS SOFTWARE IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY
    FOR THE SOFTWARE, TO THE EXTENT PERMITTED BY APPLICABLE LAW. EXCEPT
    WHEN OTHERWISE STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER
    PARTIES PROVIDE THE SOFTWARE "AS IS" WITHOUT WARRANTY OF ANY KIND,
    EITHER EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
    ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE OF THE SOFTWARE IS WITH
    YOU. SHOULD THE SOFTWARE PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL
    NECESSARY SERVICING, REPAIR, OR CORRECTION.

    IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
    WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
    REDISTRIBUTE THE SOFTWARE AS PERMITTED BY THE ABOVE LICENCE, BE LIABLE
    TO YOU FOR DAMAGES, INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL, OR
    CONSEQUENTIAL DAMAGES ARISING OUT OF THE USE OR INABILITY TO USE THE
    SOFTWARE (INCLUDING BUT NOT LIMITED TO LOSS OF DATA OR DATA BEING
    RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD PARTIES OR A
    FAILURE OF THE SOFTWARE TO OPERATE WITH ANY OTHER SOFTWARE), EVEN IF
    SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH
    DAMAGES.



Changes
Revision history for Perl extension Dist-Zilla-PluginBundle-Author-RTHOMPSON

0.142221  2014-08-10 16:38:56PDT-0700 America/Los_Angeles
    - Switch to PluginRemover for implementation of "-remove" option
    - Add the Config::Slicer role to enable arbitrary plugin options
    - Improved tests, including tests for the two above additions

0.142220  2014-08-10 15:36:20PDT-0700 America/Los_Angeles
    - Fix handling of copy_file and move_file arguments when they are
      empty. This eliminates an error on some configurations but
      otherwise has no visible effect.

0.142180  2014-08-06 10:50:49PDT-0700 America/Los_Angeles
    - Fix a few mismatches between comments/documentation and code
    - Fix a few dependencies that were specified for the wrong phase
    - Use the new release-only mode of ReadmeAnyFromPod

0.133030  2013-10-30 01:53:13 America/Los_Angeles
    - Move ConfirmRelease plugin to the end. This prevents the
    frustrating case where you answer yes to confirm the release and
    then the release aborts anyway to to a subsequent pre-release
    check.

0.132960  2013-10-23 15:39:25 America/Los_Angeles
    - Actually use the "allow_dirty" option instead of ignoring the
      user-set value and always using the default.
    - Fix broken vcs dispatch (broken since v0.132800)
    - Only load CopyFilesFromBuild plugin if it is used

0.132801  2013-10-07 16:51:23 America/Los_Angeles
    - Update SynopsisTests to Test::Synopsis

0.132800  2013-10-07 16:43:53 America/Los_Angeles
    - Remove use of given/when

0.120160  2012-01-15 18:44:05 PST8PDT
    - Rename to [@Author::RTHOMPSON]

0.120120  2012-01-12 00:18:24 PST8PDT
    - Replace KwaliteeTests with Test::Kwalitee

0.120052  2012-01-05 11:23:24 PST8PDT
    - Add CheckVersionIncrement

0.120051  2012-01-04 23:39:50 PST8PDT
    - Prevent modules in corpus directory from being indexed.
      (TODO: make this configurable)

0.120050  2012-01-04 16:04:45 PST8PDT
    - Add a simple test suite that ensures a dist can be built
      successfully with the bundle
    - Add Git::Push and Git::Remote::Check with associated options
    - Small bugfix in the code for choosing whether to provide a
      version

0.111290  2011-05-08 19:18:09 PST8PDT
    - Maunally specify Dist::Zilla prerequisites, since they are not automatically detected
    - Adjust for Dist::Zilla 4.200006
    - Add CheckPrereqsIndexed

0.103532  2010-12-19 02:36:06 EST5EDT
    - Allow Changes to be dirty when releasing (Git::Commit)
    - Make allow_dirty a configurable option

0.103531  2010-12-19 02:12:28 EST5EDT
    - Allow Changes to be dirty when releasing (Git::Check)

0.103521  2010-12-18 00:03:22 EST5EDT
    - Initial CPAN release


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.020.
use strict;
use warnings;

use 5.010;

use ExtUtils::MakeMaker ;



my %WriteMakefileArgs = (
  "ABSTRACT" => "RTHOMPSON's Dist::Zilla Configuration",
  "AUTHOR" => "Ryan C. Thompson <rct\@thompsonclan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-Author-RTHOMPSON",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::PluginBundle::Author::RTHOMPSON",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Dist::Zilla::Plugin::ArchiveRelease" => 0,
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::AutoVersion" => 0,
    "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
    "Dist::Zilla::Plugin::CheckVersionIncrement" => 0,
    "Dist::Zilla::Plugin::ConfirmRelease" => 0,
    "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
    "Dist::Zilla::Plugin::ExecDir" => 0,
    "Dist::Zilla::Plugin::ExtraTests" => 0,
    "Dist::Zilla::Plugin::GatherDir" => 0,
    "Dist::Zilla::Plugin::Git::Check" => 0,
    "Dist::Zilla::Plugin::Git::Commit" => 0,
    "Dist::Zilla::Plugin::Git::Push" => 0,
    "Dist::Zilla::Plugin::Git::Remote::Check" => 0,
    "Dist::Zilla::Plugin::Git::Tag" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => 0,
    "Dist::Zilla::Plugin::HasVersionTests" => 0,
    "Dist::Zilla::Plugin::InstallGuide" => 0,
    "Dist::Zilla::Plugin::License" => 0,
    "Dist::Zilla::Plugin::MakeMaker" => 0,
    "Dist::Zilla::Plugin::Manifest" => 0,
    "Dist::Zilla::Plugin::ManifestSkip" => 0,
    "Dist::Zilla::Plugin::MetaYAML" => 0,
    "Dist::Zilla::Plugin::MinimumPerl" => 0,
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Plugin::PkgVersion" => 0,
    "Dist::Zilla::Plugin::PodCoverageTests" => 0,
    "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::Prereqs" => 0,
    "Dist::Zilla::Plugin::PruneCruft" => 0,
    "Dist::Zilla::Plugin::PruneFiles" => 0,
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => "0.142180",
    "Dist::Zilla::Plugin::ReportVersions" => 0,
    "Dist::Zilla::Plugin::ShareDir" => 0,
    "Dist::Zilla::Plugin::StaticVersion" => 0,
    "Dist::Zilla::Plugin::Test::Compile" => 0,
    "Dist::Zilla::Plugin::Test::Kwalitee" => 0,
    "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::Plugin::Test::Synopsis" => 0,
    "Dist::Zilla::Plugin::Test::UnusedVars" => 0,
    "Dist::Zilla::Plugin::TestRelease" => 0,
    "Dist::Zilla::Plugin::UploadToCPAN" => 0,
    "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
    "Moose" => 0,
    "feature" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Moose::Autobox" => 0,
    "Scalar::Util" => 0,
    "Test::DZil" => 0,
    "Test::More" => "0.94",
    "Test::Most" => 0,
    "autodie" => 0
  },
  "VERSION" => "0.142221",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "Dist::Zilla::Plugin::ArchiveRelease" => 0,
  "Dist::Zilla::Plugin::AutoPrereqs" => 0,
  "Dist::Zilla::Plugin::AutoVersion" => 0,
  "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
  "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
  "Dist::Zilla::Plugin::CheckVersionIncrement" => 0,
  "Dist::Zilla::Plugin::ConfirmRelease" => 0,
  "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
  "Dist::Zilla::Plugin::ExecDir" => 0,
  "Dist::Zilla::Plugin::ExtraTests" => 0,
  "Dist::Zilla::Plugin::GatherDir" => 0,
  "Dist::Zilla::Plugin::Git::Check" => 0,
  "Dist::Zilla::Plugin::Git::Commit" => 0,
  "Dist::Zilla::Plugin::Git::Push" => 0,
  "Dist::Zilla::Plugin::Git::Remote::Check" => 0,
  "Dist::Zilla::Plugin::Git::Tag" => 0,
  "Dist::Zilla::Plugin::GithubMeta" => 0,
  "Dist::Zilla::Plugin::HasVersionTests" => 0,
  "Dist::Zilla::Plugin::InstallGuide" => 0,
  "Dist::Zilla::Plugin::License" => 0,
  "Dist::Zilla::Plugin::MakeMaker" => 0,
  "Dist::Zilla::Plugin::Manifest" => 0,
  "Dist::Zilla::Plugin::ManifestSkip" => 0,
  "Dist::Zilla::Plugin::MetaYAML" => 0,
  "Dist::Zilla::Plugin::MinimumPerl" => 0,
  "Dist::Zilla::Plugin::NextRelease" => 0,
  "Dist::Zilla::Plugin::PkgVersion" => 0,
  "Dist::Zilla::Plugin::PodCoverageTests" => 0,
  "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
  "Dist::Zilla::Plugin::PodWeaver" => 0,
  "Dist::Zilla::Plugin::Prereqs" => 0,
  "Dist::Zilla::Plugin::PruneCruft" => 0,
  "Dist::Zilla::Plugin::PruneFiles" => 0,
  "Dist::Zilla::Plugin::ReadmeAnyFromPod" => "0.142180",
  "Dist::Zilla::Plugin::ReportVersions" => 0,
  "Dist::Zilla::Plugin::ShareDir" => 0,
  "Dist::Zilla::Plugin::StaticVersion" => 0,
  "Dist::Zilla::Plugin::Test::Compile" => 0,
  "Dist::Zilla::Plugin::Test::Kwalitee" => 0,
  "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
  "Dist::Zilla::Plugin::Test::Portability" => 0,
  "Dist::Zilla::Plugin::Test::Synopsis" => 0,
  "Dist::Zilla::Plugin::Test::UnusedVars" => 0,
  "Dist::Zilla::Plugin::TestRelease" => 0,
  "Dist::Zilla::Plugin::UploadToCPAN" => 0,
  "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => 0,
  "Moose::Autobox" => 0,
  "Scalar::Util" => 0,
  "Test::DZil" => 0,
  "Test::More" => "0.94",
  "Test::Most" => 0,
  "autodie" => 0,
  "feature" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



